src/actions/TemperatureSyncActions.js

```diff
@@ -119,13 +120,40 @@ const downloadLogsComplete = () => ({
 });
 
 const saveLogs = (logData, sensor) => async dispatch => {
+  const { length: numberOfLogs = 0 } = logData;
```

Comment:
hooopefully my comments help with this. Essentially:

- find the most recent log that has been saved. 
- work out how long it has been since then and now in terms of the number of log intervals of the sensor. 
- count back that many logs from the the overall logs downloaded from the sensor

We were: syncing the logs and counting back from the current time, then deleting the logs. But, if the delete fails - 💣 oops